### PR TITLE
Move unnecessary force lint check to the contract analyzer

### PR DESCRIPTION
**Pull request by dsainati1**
Work on https://github.com/onflow/cadence/issues/1269

## Description
This moves the unnecessary force check to the new contract analyzer
______

- [X] Targeted PR against `master` branch
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [X] Updated relevant documentation 
- [X] Re-reviewed `Files changed` in the Github PR explorer
- [X] Added appropriate labels

### CODE REVIEW

File: tools/contract-analyzer/go.mod [hunk 1]
```diff
@@ -123,3 +123,5 @@ require (
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
 	lukechampine.com/blake3 v1.1.7 // indirect
 )
+
+replace github.com/onflow/cadence => ../..
```
Comment (dsainati1) [inline tools/contract-analyzer/go.mod:127]: Until we release a new version of Cadence with the changes that the contract analyzer will need, we can just use the local version of Cadence for the analyzer

File: runtime/sema/elaboration.go [hunk 2]
```diff
@@ -145,6 +146,7 @@ func NewElaboration(gauge common.MemoryGauge) *Elaboration {
 		ReferenceExpressionBorrowTypes:      map[*ast.ReferenceExpression]Type{},
 		IndexExpressionIndexedTypes:         map[*ast.IndexExpression]ValueIndexableType{},
 		IndexExpressionIndexingTypes:        map[*ast.IndexExpression]Type{},
+		ForceExpressionTypes:                map[*ast.ForceExpression]Type{},
```
Comment (turbolent) [inline runtime/sema/elaboration.go:149]: Maybe also gate this behind `if checker.lintingEnabled {`, so we don't have to unnecessarily allocate it

File: tools/contract-analyzer/analyzers/analyzers_test.go [hunk 3]
```diff
@@ -135,3 +135,58 @@ func TestReferenceOperatorAnalyzer(t *testing.T) {
 		diagnostics,
 	)
 }
+
+func TestForceOperatorAnalyzer(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("unnecessary", func(t *testing.T) {
+		diagnostics := testAnalyzers(t,
```
Comment (turbolent) [inline tools/contract-analyzer/analyzers/analyzers_test.go:144]:
```suggestion
		t.Parallel()

		diagnostics := testAnalyzers(t,
```

File: tools/contract-analyzer/analyzers/analyzers_test.go [hunk 4]
```diff
@@ -135,3 +135,58 @@ func TestReferenceOperatorAnalyzer(t *testing.T) {
...
+					Category: "lint",
+					Message:  "unnecessary force operator",
+				},
+			},
+			diagnostics,
+		)
+	})
+
+	t.Run("valid", func(t *testing.T) {
+		diagnostics := testAnalyzers(t,
```
Comment (turbolent) [inline tools/contract-analyzer/analyzers/analyzers_test.go:174]:
```suggestion
	t.Run("valid", func(t *testing.T) {
		t.Parallel()
	
		diagnostics := testAnalyzers(t,
```

File: tools/contract-analyzer/go.mod (hunk 1 above)
Comment (turbolent) [reply tools/contract-analyzer/go.mod:127]: Once this PR is merged, can you please open a follow-up PR which removes this and updated to the latest commit of the `master` branch?

**Review by turbolent — APPROVED** [review]

File: tools/contract-analyzer/go.mod (hunk 1 above)
Comment (dsainati1) [reply tools/contract-analyzer/go.mod:127]: Yea, I will once all of the linting rules have been migrated over. Just thought I'd do them in separate PRs so they are easier to review.

**Pull request merged**